### PR TITLE
util/log: combine --alsologtostderr and --log-threshold

### PR DESCRIPTION
**Pull request by petermattis**
The --alsologtostderr flag is equivalent to --log-threshold=INFO. Moved
the severity functionality from --log-threshold to --alsologtostderr.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4820)

### CODE REVIEW

File: util/log/clog.go [hunk 1]
```diff
@@ -132,6 +132,12 @@ func SeverityByName(s string) (Severity, bool) {
 			return Severity(i), true
 		}
 	}
+	switch s {
+	case "TRUE":
+		return InfoLog, true
+	case "FALSE":
+		return NumSeverity, true
```
Comment (tschottdorf) [inline util/log/clog.go]: should NumSeverity be renamed?

**Comment by tschottdorf** [comment]
LGTM

File: util/log/clog.go (hunk 1 above)
Comment (petermattis) [inline util/log/clog.go]: Maybe. Doesn't seem too harmful to leave it like this.

**Comment by jseldess** [comment]
@petermattis, since we write to logs and stderr by default, the `--alsologtostderr` label seems confusing to me. Would it be clearer to keep `--log-threshold` instead?

**Comment by petermattis** [comment]
We write to stderr by default because --alsologtostderr defaults to ERROR.
I like the verbosity of that flag name because it makes clear we are
logging to stderr in addition to somewhere else. The --log-threshold flag
name obscured the fact that flag only applied to stderr.

**Comment by jseldess** [comment]
OK, thanks for clarifying. What do you think about having `--also-log-to-stderr` and `--only-log-to-stderr` then? Might make those two flags easier to contrast.

**Comment by petermattis** [comment]
The names are historical and likely only to be used by CockroachDB devs
(not users). I don't think it is worth changing them.

**Comment by jseldess** [comment]
Got it. Added to docs in https://github.com/cockroachdb/docs/pull/88

**Comment by bdarnell** [comment]
I don't think these flags are developer-only. Docker containers are expected to log to stdout/stderr instead of to disk, as are apps deployed with the http://12factor.net model. We've also diverged significantly from the google/glog model already, to the point that I think keeping those names confuses more than it helps. 

I don't want to bikeshed endlessly on this, but I suggest we get rid of one of the stderr-related flags:
* `--log-dir` accepts a special value (`none`? `/dev/null`?) to disable logging to disk, instead of the unrelated-looking `--logtostderr`. Or we make `--log-dir` more structured (as with `--store`) to give us room to add support for syslog, etc in the future.
* Rename `--alsologtostderr[=level]` to `--log-stderr=level` and remove the weird semi-boolean `NoOptDefValue` (the `--help` output for that is really confusing)

**Comment by petermattis** [comment]
I'm fine with simplifying the logging flags some more, though I feel I've done my part for the time being. If you'd like to jump in with your suggestions I think they are reasonable and reasonably straightforward to implement. The only one I'd be hesitant to do is to extend `--log-dir` with additional structure. A special value such as `none` or `/dev/null` or `-` seems better.

**Pull request merged**